zjit/src/cruby.rs

```diff
@@ -1218,6 +1218,14 @@ pub fn get_class_name(class: VALUE) -> String {
     }).unwrap_or_else(|| "Unknown".to_string())
 }
 
+pub fn class_has_safe_allocator(class: VALUE) -> bool {
+    if class == unsafe { rb_cHash } { return true; }
```

Comment:
Could you maybe add a comment for each non-default-allocator class that mentions the name of the allocator function? I don't know where's the allocator function for Hash/Array/String/Regexp and I'm not sure why they're safe, so I would like to look at them.